src/main/java/edu/harvard/iq/dataverse/DataverseSession.java

```diff
@@ -210,6 +221,9 @@ public void dismissMessage(BannerMessage message){
     }
     
     public void configureSessionTimeout() {
+        if (user instanceof GuestUser) {
```

Comment:
Could you explain what this adds?

Reply:
Right. This is the thing @landreev and I talked about in Slack: https://iqss.slack.com/archives/C010LA04BCG/p1613509511062000

In short, when `configureSessionTimeout` was added, it was always put in after `setUser`, with the implication that we're now dealing with an AuthenticatedUser (or at least a non-GuestUser).

That is to say, all over the code, you'll see these two lines execute one after the other:

```
setUser(user);
configureSessionTimeout();
```

Prior to this pull request `configureSessionTimeout` could assume it has a non-GuestUser and would go ahead an lengthen the session. Now we can't make that assumption since `setUser` returns early if the user is disabled, with the result that the session contains a GuestUser rather than an AuthenticatedUser.

Reply:
I just checked and see what you're saying; we *always* call configureSessionTimeout after setUser. Why don't we just move that logic into setUser then? (and it would be after the early return). We can then not have to remember to call it if we add a setUser again somewhere else.

Reply:
@landreev do you have any objection to what @scolapasta is proposing above?

Reply:
I don't remember much about that code (or what exactly I was thinking when I implemented it that way). But yes, it looks like it would be possible - to just call it from inside setUser().

Reply:
P.S. as long as we make sure it's called from inside this `if` block in setUser(): 
```
FacesContext context = FacesContext.getCurrentInstance();
if(context != null) {
   ...
}
```
... or otherwise in a way that's not going to result in a null pointer if it's called from a place where faces context is unavailable...

Reply:
Thanks, all. Fixed in 610e9ca32.